src/test/scala/org/labrad/ServerTest.scala

```diff
@@ -234,6 +235,19 @@ class ServerTest extends FunSuite with AsyncAssertions {
     assert(keys.contains("a"))
     assert(!keys2.contains("a"))
   }
+
+  test("server gets context expiration message when client disconnects") { fix =>
+    val ctx = withClient(fix.host, fix.port, fix.password) { client2 =>
+      val ctx = client2.newContext.copy(high = client2.id)
+      await(client2.send("Scala Test Server", ctx, "Set" -> Cluster(Str("a"), Str("test"))))
+      ctx
+    }
+    Thread.sleep(1000)
```

Comment:
you are right, and I should know better as I've certainly been burned by this before :)

Changed to use a queue to indicate which contexts have been expired.